File: tourism-system-java/src/main/java/com/tourism/model/entity/TravelDiary.java
package com.tourism.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("travel_diary")
public class TravelDiary {

    @TableId(type = IdType.INPUT)
    private String id;

    private String title;

    @TableField(value = "`content`")
    private String content;

    private String placeId;

    private Long authorId;

    private Integer clickCount;

    private BigDecimal rating;

    private Integer ratingCount;

    /** 图片URL列表，JSON数组 */
    private String images;

    /** 视频URL列表，JSON数组 */
    private String videos;

    /** 标签，JSON数组 */
    private String tags;

    @TableLogic
    private Integer deleted;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: TourismSystemApp/TourismSystemApp/Models/TravelDiary.swift
import Foundation

struct TravelDiary: Identifiable, Codable, Hashable {
    let id: String
    let title: String?
    let content: String?
    let placeId: String?
    let authorId: Int64?
    let clickCount: Int?
    let rating: Double?
    let ratingCount: Int?
    let images: String?
    let videos: String?
    let tags: String?
    let createdAt: String?
    let updatedAt: String?

    var parsedImages: [String] { parseJSONArray(images) }
    var parsedVideos: [String] { parseJSONArray(videos) }
    var parsedTags: [String] { parseJSONArray(tags) }

    func hash(into hasher: inout Hasher) { hasher.combine(id) }
    static func == (lhs: TravelDiary, rhs: TravelDiary) -> Bool { lhs.id == rhs.id }
}
